MapboxNavigation/URLSession.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import MapboxDirections
+
+extension URLSession {
+    /// The user agent string for any HTTP requests performed directly within this library.
+    static let userAgent: String = {
+        var components: [String] = []
+        
+        if let appName = Bundle.main.object(forInfoDictionaryKey: "CFBundleName") as? String ?? Bundle.main.object(forInfoDictionaryKey: "CFBundleIdentifier") as? String {
+            let version = Bundle.main.object(forInfoDictionaryKey:"CFBundleShortVersionString") as? String ?? ""
+            components.append("\(appName)/\(version)")
+        }
+        
+        let libraryBundle: Bundle? = Bundle(for: ImageDownloader.self)
+        
+        if let libraryName = libraryBundle?.object(forInfoDictionaryKey: "CFBundleName") as? String, let navVersion = libraryBundle?.object(forInfoDictionaryKey:"CFBundleShortVersionString") as? String {
+            components.append("\(libraryName)/\(navVersion)")
+        }
+        
+        let directionsLibraryBundle: Bundle? = Bundle(for: Directions.self)
+        
+        if let directionsLibraryName = directionsLibraryBundle?.object(forInfoDictionaryKey: "CFBundleName") as? String, let directionsVersion = directionsLibraryBundle?.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String {
```

Comment:
This is a good reminder of the importance of updating `CFBundleShortVersionString` using `agvtool` whenever we release a new version of a library. 👍